### PR TITLE
Add MobileNetV2 models

### PR DESCRIPTION
**Pull request by Rishit-dagli**
# What does this PR do?

For completeness following #526 this PR ports MobileNetV2 as well.

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: keras_cv/models/mobilenet_v2.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+        a function that takes an input Tensor representing a Depth layer.
+    """
+    if name is None:
+        name = f"depth_{backend.get_uid('depth')}"
+
+    if min_value is None:
+        min_value = divisor
+
+    def apply(x):
+        new_x = max(min_value, int(x + divisor / 2) // divisor * divisor)
```
Comment (LukeWood) [inline keras_cv/models/mobilenet_v2.py:88]: is depth a layer?

File: keras_cv/models/mobilenet_v2.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+            "If `include_top` is True, "
+            "you should specify `num_classes`. "
+            f"Received: num_classes={num_classes}"
+        )
+
+    input = layers.Input(shape=input_shape)
+    x = None
+
+    if include_rescaling:
+        x = layers.Rescaling(scale=1.0 / 127.5, offset=-1.0)(input)
```
Comment (LukeWood) [inline keras_cv/models/mobilenet_v2.py:291]: We should normalize by 1/255

**Review by LukeWood — CHANGES_REQUESTED** [review]
Minor changes, thanks for the PR!

File: keras_cv/models/mobilenet_v2.py (hunk 2 above)
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v2.py:291]: The original `keras.applications` uses this rescaling in the preprocessing step, please see:

https://github.com/keras-team/keras/blob/v2.9.0/keras/applications/mobilenet_v2.py#L517

https://github.com/keras-team/keras/blob/07e13740fd181fc3ddec7d9a594d8a08666645f6/keras/applications/imagenet_utils.py#L191-L194

Thus, I thought to go ahead with this

File: keras_cv/models/mobilenet_v2.py (hunk 1 above)
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v2.py:88]: Rewritten as in #526

File: keras_cv/models/mobilenet_v2.py (hunk 2 above)
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v2.py:287]: Considering https://github.com/keras-team/keras-cv/pull/526#discussion_r923807778 , I will use 1/255 for this

**Comment by tanzhenyu** [comment]
@ianstenbit Is MobileNetV2 already supported?

**Comment by ianstenbit** [comment]
> @ianstenbit Is MobileNetV2 already supported?

No, right now we only have MobileNetV3, and I think something may be wrong with the implementation because I cannot get the model to train to anywhere near the paper-claimed result.

**Comment by tanzhenyu** [comment]
> > @ianstenbit Is MobileNetV2 already supported?
> 
> No, right now we only have MobileNetV3, and I think something may be wrong with the implementation because I cannot get the model to train to anywhere near the paper-claimed result.

Got it. @Rishit-dagli can you try running this model with our existing [training script](https://github.com/keras-team/keras-cv/blob/master/examples/training/classification/imagenet/basic_training.py) and let us know how the metrics look like, for maybe 10-30 epochs? This would help us verify the model to some extent, and we can run the model end to end

**Comment by LukeWood** [comment]
Closing due to inactivity. 

Thanks for your enthusiasm, feel free to reply if you'd like to work on this again.
